### ISSUE TITLE
Add configurable server port

### ISSUE THREAD

**Issue by vaidotas-adf** [garris/BackstopJS#130]
BackstopJS report server runs on same port as [Browsersync](https://www.browsersync.io/) UI (port 3001). It would be nice to have configurable browser port in config file.
Alternative solution: detect if port is free, and use another one, if default port is already taken.

**Comment by garris** [comment]
Hi,  thanks for posting.  I will put up a fix for this.   Will let you know when it's up so you can test it out.  Best, GS

**Comment by garris** [comment]
I added a property in the default config to assign a port number.  Please do a fresh install from master and let me know if that works for you.   Cheers.

**Comment by vaidotas-adf** [comment]
I've set "port": 3002 in config. When starting console shows "Listening on port: 3002". server.js process is started with -p 3002 parameter. But trying to open http://localhost:3002/compare/ shows that page is not available. Somehow server is still started on 3001 port.

**Comment by garris** [comment]
Thanks for testing this.  I am seeing very weird `argv` behavior that was not there when adding the `-t` parameter.  Perhaps this is an issue with `argv` & `spawn` together?  Will follow up very soon as this is something I think we should have.  Please let me know if you gain any insight in the meantime.  Thank you.

**Comment by vaidotas-adf** [comment]
Looks like that separate arguments need to be passed as separate array elements to spawn function. Starting server on custom port works correctly after modifying line 33 in /gulp/tasks/start.js file:

    var serverHook = spawn('node', ['server.js', '-t', time, '-p', port],  {detached: true, stdio:'ignore'});

**Comment by garris** [comment]
Nice,  Thank you!   Merged and tested.   This is now available on master and will go out in the next release.    Cheers.

**Comment by LukasLt** [comment]
Is it possible to add check before starting to know if selected port is taken, and if so - give any error message?

I just had the same issue with browser sync (I totally forgot it was running form yesterday), and despite doing it on fresh install I got no success. For me it was was pure luck to notice that the issue is taken port and not some broken dependency causing this.

Thanks!
